Library/Homebrew/language/python.rb

```diff
@@ -228,9 +231,32 @@ def virtualenv_install_with_resources(using: nil, system_site_packages: true, wi
           python = T.must(wanted.first)
           python = "python3" if python == "python"
         end
+
+        venv_resources = if without.nil? && start_with.nil? && end_with.nil?
+          resources
+        else
+          remaining_resources = resources.to_h { |resource| [resource.name, resource] }
+          if (unknown_resource = Array(without).find { |r| remaining_resources.delete(r).nil? })
+            raise ArgumentError, "#{__method__} `without` includes unknown resource \"#{unknown_resource}\""
+          end
+
+          start_with = Array(start_with)
+          start_with_resources = start_with.map { |r| remaining_resources.delete(r) }
+          unless (index = start_with_resources.find_index(nil)).nil?
+            raise ArgumentError, "#{__method__} `start_with` includes unknown resource \"#{start_with[index]}\""
+          end
+
+          end_with_resources = Array(end_with).map do |r|
+            remaining_resources.delete(r) ||
+              raise(ArgumentError, "#{__method__} `end_with` includes unknown resource \"#{r}\"")
+          end
```

Comment:
Maybe I should just loop it:
```suggestion
          _without_resources, start_with_resources, end_with_resources = [without, start_with, end_with].map do |arg|
            Array(arg).map do |r|
              remaining_resources.delete(r) ||
                raise(ArgumentError, "#{__method__} has an argument with unknown resource \"#{r}\"")
            end
          end
```

If so, not sure if `||` or `do ... end` exception handling is better, e.g.
```ruby
          _without_resources, start_with_resources, end_with_resources = [without, start_with, end_with].map do |arg|
            Array(arg).map do |r|
              remaining_resources.delete(r) do
                raise ArgumentError, "#{__method__} has an argument with unknown resource \"#{r}\""
              end
            end
          end
```

Reply:
Looks like Sorbet can't tell all entries assignments are non-nil. May experiment with some other ways to see if this can work.